e2e/e2e_test.go

```
@@ -0,0 +1,166 @@
+package e2e_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+
+	"dagger.io/dagger"
+	"github.com/spf13/cobra"
+)
+
+var runTests = &cobra.Command{
+	Use:   "test",
+	Short: "run k6 tests",
+	Run:   e2eTests,
+}
+
+var errorHandling = &cobra.Command{
+	Use:   "error handling",
+	Short: "check if k6 tests have failing tests",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if testFailures > 0 {
+			return fmt.Errorf("Failing tests")
+		}
+		return nil
+	},
+}
+
+var testFailures = 0
+
+func e2eTests(cmd *cobra.Command, args []string) {
+	ctx := cmd.Context()
+
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr), dagger.WithWorkdir(".."))
+	if err != nil {
+		panic(err)
```

Instead of panicking, we should return errors

---

When changing this, make sure to update all the places errors were ignored